src/main/scala/com/fulcrumgenomics/util/SampleBarcodeMetric.scala

```diff
@@ -45,78 +45,116 @@ object SampleBarcodeMetric {
+        val fracPfMatches = metric.pf_templates / totalPfReads.toDouble
+        if (fracPfMatches > bestPfPctOfAllBarcodeMatches) {
+          bestPfPctOfAllBarcodeMatches = fracPfMatches
         }
-        metric.pf_pct_matches = pctPfMatches
+        metric.pf_frac_matches = fracPfMatches
       }
       if (bestPfPctOfAllBarcodeMatches > 0) {
-        noMatchMetric.pf_ratio_this_barcode_to_best_barcode_pct = noMatchMetric.pf_pct_matches / bestPfPctOfAllBarcodeMatches
+        noMatchMetric.pf_ratio_this_barcode_to_best_barcode_frac = noMatchMetric.pf_frac_matches / bestPfPctOfAllBarcodeMatches
         barcodeToMetrics.foreach { case (_, metric) =>
-          metric.pf_ratio_this_barcode_to_best_barcode_pct = metric.pf_pct_matches / bestPfPctOfAllBarcodeMatches
+          metric.pf_ratio_this_barcode_to_best_barcode_frac = metric.pf_frac_matches / bestPfPctOfAllBarcodeMatches
         }
       }
 
     }
     if (totalPfReadsAssigned > 0) {
       val mean: Double = totalPfReadsAssigned.toDouble / barcodeToMetrics.values.size.toDouble
       barcodeToMetrics.foreach { case (barcode, metric) =>
-        metric.pf_normalized_matches = metric.pf_reads / mean
+        metric.pf_normalized_matches = metric.pf_templates / mean
       }
     }
   }
 }
 
 /**
-  * Metrics for matching reads to sample barcodes.
+  * Metrics for matching templates to sample barcodes primarily used in [[com.fulcrumgenomics.fastq.DemuxFastqs]].
+  *
+  * The number of templates will match the number of reads for an Illumina single-end sequencing run, while the number
+  * of templates will be half the number of reads for an Illumina paired-end sequencing run (i.e. R1 & R2 observe the
+  * same template).
+  *
+  * @param barcode_name the name for the sample barcode, typically the sample name from the SampleSheet.
+  * @param library_name the name of the library, typically the library identifier from the SampleSheet.
+  * @param barcode the sample barcode bases.  Dual index barcodes will have two sample barcode sequences delimited by a
+  *                dash.
+  * @param templates the total number of templates matching the given barcode.
+  * @param pf_templates the total number of pass-filter templates matching the given barcode.
+  * @param perfect_matches the number of templates that match perfectly the given barcode.
+  * @param pf_perfect_matches the number of pass-filter templates that match perfectly the given barcode.
+  * @param one_mismatch_matches the number of pass-filter templates that match the given barcode with exactly one
+  *                             mismatch.
+  * @param pf_one_mismatch_matches the number of pass-filter templates that match the given barcode with exactly
+  *                                one mismatch.
+  * @param frac_matches the fraction of all templates that match the given barcode.
+  * @param ratio_this_barcode_to_best_barcode_frac the rate of all templates matching this barcode to all template
```

Comment:
I don't know why this ever acquired a `_pct` suffix since it's a _ratio_.  Silly Picard.  I'd drop the `_frac_ suffix here and on the other ratio.